asset_dashboard/models.py

```diff
@@ -64,33 +64,42 @@ def __str__(self):
 class ProjectScore(models.Model):
 
     project = models.OneToOneField(Project, on_delete=models.CASCADE)
-    core_mission_score = models.IntegerField(default=1,
+    core_mission_score = models.IntegerField(null=True, blank=True,
                                              validators=[MinValueValidator(1),
                                                          MaxValueValidator(5)])
-    operations_impact_score = models.IntegerField(default=1,
+    operations_impact_score = models.IntegerField(null=True, blank=True,
                                                   validators=[MinValueValidator(1),
                                                               MaxValueValidator(5)])
-    sustainability_score = models.IntegerField(default=1,
+    sustainability_score = models.IntegerField(null=True, blank=True,
                                                validators=[MinValueValidator(1),
                                                            MaxValueValidator(5)])
-    ease_score = models.IntegerField(default=1,
+    ease_score = models.IntegerField(null=True, blank=True,
                                      validators=[MinValueValidator(1),
                                                  MaxValueValidator(5)])
-    geographic_distance_score = models.IntegerField(default=1,
+    geographic_distance_score = models.IntegerField(null=True, blank=True,
                                                     validators=[MinValueValidator(1),
                                                                 MaxValueValidator(5)])
-    social_equity_score = models.IntegerField(default=1,
+    social_equity_score = models.IntegerField(null=True, blank=True,
                                               validators=[MinValueValidator(1),
                                                           MaxValueValidator(5)])
 
     @property
     def total_score(self):
-        '''return the total, weighted score'''
-        ...
+        # get the ProjectScore fields, excluding the id and project fields
+        score_fields = [f for f in self._meta.get_fields() if f.name != 'id' if f.name != 'project']
```

Comment:
this is a good solution!

i think we have an opportunity to do something a bit safer and much clearer.

let's define a custom fieldtype that looks like

```python
class ScoreField(models.IntegerField):

     def __init__(self, *args, **kwargs):
          if 'null' not in kwargs:
              kwargs['null'] = True
          if 'blank' not in kwargs:
               kwargs['blank'] = True
          if 'validators' not in kwargs:
               validators = [MinValueValidator(1), MaxValueValidator(5)]
          super().__init(self, *args, **kwargs)

class ProjectScore(models.Model):

     ...
     core_mission_score = ScoreField()
     ...

     @property
     def total_score(self):
          score_fields = [f for f in self._meta.get_fields() if type(f) == ScoreField]
          ...
```

Does that make sense?

Reply:
real cool, i'm about to try it. thanks.

Reply:
Good DRY code at this new commit: 8822d65

Reply:
nice!